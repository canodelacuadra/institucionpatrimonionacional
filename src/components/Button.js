// src/components/Button.js
import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ text, onClick, style }) => {
    return (
        <button className="button" onClick={onClick} style={style}>
            {text}
        </button>
    );
};

Button.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    style: PropTypes.object,
};

Button.defaultProps = {
    onClick: () => { },
    style: {},
};

export default Button;
