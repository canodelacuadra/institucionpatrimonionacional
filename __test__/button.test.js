// src/components/__tests__/Button.test.js
import React from 'react';
import { render } from '@testing-library/react';
import Button from '../src/components/Button';

test('renderiza el Button correctamente y coincide con el snapshot', () => {
    const { asFragment } = render(<Button text="Click Me" style={{ backgroundColor: 'blue', color: 'white' }} />);
    expect(asFragment()).toMatchSnapshot();
});
