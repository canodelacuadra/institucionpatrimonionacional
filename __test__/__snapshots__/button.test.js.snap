// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza el Button correctamente y coincide con el snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: blue; color: white;"
  >
    Click Me
  </button>
</DocumentFragment>
`;
